### PR TITLE
Handle env vars in a case-preserving, case-insensitive manner on Windows

### PR DESCRIPTION
**Pull request by niik**
This is a follow-up to #570 and attempts to tackle Windows environment variable shenanigans for all environment variables and not just `PATH`. The main change is the introduction of `EnvMap`, a class that behaves as a case-insensitive, case-preserving map on Windows and a regular case-sensitive map on other platforms. This class is used in `lib/git-environment.ts` to handle environment variables in a consistent manner across platforms.

I've also bumped Prettier to 3.3.1 so it can handle my syntax and specified the version of TypeScript used by Visual Studio Code in `.vscode/settings.json` because I was building this on an old VM with an old VS code and noticed that it wouldn't complete properly.

### CODE REVIEW

**Comment by tidy-dev** [comment]
@niik Just making sure you were aware of the failing tests (I have been putting off reviewing till those are happy)

**Comment by niik** [comment]
> Just making sure you were aware of the failing tests (I have been putting off reviewing till those are happy)

I was not aware, thanks for the heads up!

**Review by sergiou87 — COMMENTED** [review]
It looks all good! Just have a question about one of the tests, not sure if it's 100% useful the way it is right now 🤔

File: test/fast/environment-test.ts [hunk 1]
```diff
@@ -53,11 +53,20 @@ describe('environment variables', () => {
         // case-insensitive (like Windows) we don't end up with an invalid PATH
         // and the original one lost in the process.
         const { env } = setupEnvironment({})
-        expect(env.PATH).toContain('wow-such-case-insensitivity')
+        expect(env.Path).toContain('wow-such-case-insensitivity')
```
[the remark this answers is not in the document]
Comment (niik) [reply test/fast/environment-test.ts]: Oh boy, this was 5 months ago but I _think_ this is correct albeit highly confusing. It's meant to test that on Windows the env map is case _preserving_ but case insensitive. That said I tried to update the test to be more clear and found a bug in EnvMap's constructor. Happy for you to take another look now!

**Review by sergiou87 — APPROVED** [review]

**Pull request merged**